From e837e43e7d5b3254ec833abb1cc0858b93e923c6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Wed, 13 Oct 2021 22:10:51 +0200
Subject: [PATCH] ntdll: Do not memset huge arenas, they should be zeroed out
 already.

They are allocated with NtAllocateVirtualMemory directly, which maps
zeroed pages.

CW-Bug-Id: #16549
---
 dlls/ntdll/heap_lfh.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/ntdll/heap_lfh.c b/dlls/ntdll/heap_lfh.c
index 11111111111..11111111111 100644
--- a/dlls/ntdll/heap_lfh.c
+++ b/dlls/ntdll/heap_lfh.c
@@ -925,7 +925,7 @@ static FORCEINLINE LFH_ptr *LFH_allocate(ULONG flags, size_t size)
     {
         arena = LFH_allocate_huge_arena(heap, class_size);
         if (arena) block = LFH_arena_get_block(arena, ARENA_HEADER_SIZE);
-        if (block) LFH_block_initialize(block, flags, 0, size, LFH_block_get_class_size(block));
+        if (block) LFH_block_initialize(block, flags & ~HEAP_ZERO_MEMORY, 0, size, LFH_block_get_class_size(block));
     }
 
     LFH_deallocated_cached_arenas(heap);
